File: app-python/entitlements_app/requests/set_account_override_request.py
"""Pydantic request model for command `SetAccountOverride`."""

from pydantic import BaseModel


class SetAccountOverrideRequest(BaseModel):
    accountId: str
    allowance: int
    createdAt: str
    expiresAt: str | None = None
    featureKey: str
    overrideId: str
    overrideType: str
    reason: str
    tenantId: str

    @property
    def account_id(self) -> str:
        return self.accountId

    @property
    def created_at(self) -> str:
        return self.createdAt

    @property
    def expires_at(self) -> str:
        return self.expiresAt

    @property
    def feature_key(self) -> str:
        return self.featureKey

    @property
    def override_id(self) -> str:
        return self.overrideId

    @property
    def override_type(self) -> str:
        return self.overrideType

    @property
    def tenant_id(self) -> str:
        return self.tenantId

File: app/src/main/java/com/example/spring/SetAccountOverrideRequest.java
package com.example.spring;

/**
 * Request DTO for command SetAccountOverride.
 */
public record SetAccountOverrideRequest(
    @jakarta.validation.constraints.NotBlank String accountId,
    int allowance,
    @jakarta.validation.constraints.NotBlank String createdAt,
    String expiresAt,
    @jakarta.validation.constraints.NotBlank String featureKey,
    @jakarta.validation.constraints.NotBlank String overrideId,
    @jakarta.validation.constraints.NotBlank String overrideType,
    @jakarta.validation.constraints.NotBlank String reason,
    @jakarta.validation.constraints.NotBlank String tenantId
) {}
